tests/test_service_account.py

```diff
@@ -601,19 +601,23 @@ def test_authorize_stale_token(self, utcnow):
 
     @mock.patch('oauth2client.client._UTCNOW')
     def test_authorize_401(self, utcnow):
+        # Test that a 401 triggers a refresh of the access token.
         utcnow.return_value = T1_DATE
 
+        # First request should use the first access token.
+        # Second request returns 401 and should refresh the access token.
+        # Third request should use the second access token.
         http = http_mock.HttpMockSequence([
             ({'status': http_client.OK}, b''),
             ({'status': http_client.UNAUTHORIZED}, b''),
             ({'status': http_client.OK}, b''),
         ])
-        self.jwt.authorize(http)
-        transport.request(http, self.url)
+        http = self.jwt.authorize(http)
+        http.request(self.url)
```

Comment:
(That is to say, the interface for transport.request is still in flux, so sit tight)